pkg/monitor/monitor.go

```diff
@@ -136,35 +167,65 @@ func (monitor *controllerMonitor) createControllerCRD() (*v1beta1.AntreaControll
 	return monitor.client.ClusterinformationV1beta1().AntreaControllerInfos().Create(controllerCRD)
 }
 
-// TODO: Update network policy related fields when the upstreaming is ready
+// updateControllerCRD updates all the fields of existing monitoring CRD
 func (monitor *controllerMonitor) updateControllerCRD(controllerCRD *v1beta1.AntreaControllerInfo) (*v1beta1.AntreaControllerInfo, error) {
+	controllerCRD.Version = version.GetFullVersion()
+	controllerCRD.PodRef = monitor.GetSelfNode()
+	controllerCRD.NodeRef = monitor.GetSelfNode()
+	controllerCRD.ServiceRef = monitor.GetService()
+	controllerCRD.ControllerConditions = []v1beta1.ControllerCondition{
+		{
+			Type:              v1beta1.ControllerHealthy,
+			Status:            v1.ConditionTrue,
+			LastHeartbeatTime: metav1.Now(),
+		},
+	}
 	klog.V(2).Infof("Updating controller monitor CRD %v", controllerCRD)
+	return monitor.client.ClusterinformationV1beta1().AntreaControllerInfos().Update(controllerCRD)
+}
+
+// partialUpdateControllerCRD only updates the variables.
+func (monitor *controllerMonitor) partialUpdateControllerCRD(controllerCRD *v1beta1.AntreaControllerInfo) (*v1beta1.AntreaControllerInfo, error) {
 	controllerCRD.ControllerConditions = []v1beta1.ControllerCondition{
 		{
 			Type:              v1beta1.ControllerHealthy,
 			Status:            v1.ConditionTrue,
 			LastHeartbeatTime: metav1.Now(),
 		},
 	}
+	klog.V(2).Infof("Partially updating controller monitor CRD %v", controllerCRD)
 	return monitor.client.ClusterinformationV1beta1().AntreaControllerInfos().Update(controllerCRD)
 }
 
+func (monitor *controllerMonitor) deleteAgentCRD(old interface{}) {
+	node := old.(*v1.Node)
+	// No dependents for AntreaAgentInfo so no cascading deletion needed.
+	deletePolicy := metav1.DeletePropagationOrphan
+	deleteOptions := &metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	}
```

Comment:
Do we need to set the option explicitly? I think it's required only when there're relationships existing? otherwise the option might confuse people to think we are intending to keep its dependents while it actually doesn't have any. 
I didn't see this option is used in deletion of Endpoint, Node, Pod, and saw it in DaemonSet and Deployment which do have dependents.

Reply:
Thanks for the comment. We can set an empty DeleteOption here. I will update this part.